update_entrydns.sh

```diff
@@ -1,12 +1,28 @@
-#!/bin/sh
+#!/bin/bash
 
-[ -z "${AUTH_TOKEN}" ] && { echo "=> AUTH_TOKEN cannot be empty" && exit 1; }
+([ -z "${AUTH_TOKEN}" ] && [ -z "${AUTH_TOKENS}" ]) && { echo "=> AUTH_TOKEN(S) cannot be empty" && exit 1; }
 
-IPADDRESS=`curl -s icanhazip.com`
-if [[ "${IPADDRESS}" != "$(cat /current_ip)" ]]; then
+updateIP() {
+  local token="$1"
   echo "IP does not match, updating DNS"
   echo ${IPADDRESS} >|/current_ip
-  curl -k -X PUT -d "ip=${IPADDRESS}" "https://entrydns.net/records/modify/${AUTH_TOKEN}"
+  curl -k -X PUT -d "ip=${IPADDRESS}" "https://entrydns.net/records/modify/$token"  
+}
+
+IPADDRESS=`curl -s icanhazip.com`
+if [[ "${IPADDRESS}" != "$(cat /current_ip)" ]]; then
+
+  if [[ "${AUTH_TOKENS}" ]]; then
+    if [ -n "${AUTH_TOKENS}" ]; then
+        declare part
+        while read -d "," part; do
```

Comment:
Wouldn't this work on a single token too (no commas)? Then you don't have to treat them as different cases. Does the while loop work without bash?